src/H5ESint.c

```diff
@@ -369,6 +553,91 @@ H5ES__handle_fail(H5ES_t *es, H5ES_event_t *ev)
     FUNC_LEAVE_NOAPI(SUCCEED)
 } /* end H5ES__handle_fail() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5ES__op_complete
+ *
+ * Purpose:     Handle an operation completing
+ *
+ * Return:      SUCCEED / FAIL
+ *
+ * Programmer:	Quincey Koziol
+ *	        Friday, December 11, 2020
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5ES__op_complete(H5ES_t *es, H5ES_event_t *ev, H5VL_request_status_t ev_status)
+{
+    hid_t  err_stack_id = H5I_INVALID_HID; /* Error stack for failed operation */
+    herr_t ret_value    = SUCCEED;         /* Return value */
+
+    FUNC_ENTER_STATIC
+
+    /* Sanity check */
+    HDassert(es);
+    HDassert(ev);
+
+    /* Check for event that hasn't actually completed */
+    if (H5VL_REQUEST_STATUS_IN_PROGRESS == ev_status || H5VL_REQUEST_STATUS_CANT_CANCEL == ev_status)
+        HGOTO_ERROR(H5E_EVENTSET, H5E_CANTSET, FAIL, "event hasn't completed?!?")
+
+    /* Handle each form of event completion */
+    if (H5VL_REQUEST_STATUS_SUCCEED == ev_status || H5VL_REQUEST_STATUS_CANCELED == ev_status) {
+        /* Invoke the event set's 'complete' callback, if present */
+        if (es->comp_func) {
+            H5ES_status_t op_status; /* Status for complete callback */
+
+            /* Set appropriate info for callback */
+            if (H5VL_REQUEST_STATUS_SUCCEED == ev_status) {
+                /* Translate status */
+                op_status = H5ES_STATUS_SUCCEED;
+
+                /* Retrieve the execution time info */
+                if (H5VL_request_specific(ev->request, H5VL_REQUEST_GET_EXEC_TIME, &ev->op_info.op_exec_ts,
```

Comment:
We're concerned that this code will add unnecessary overhead to every operation. It will also require all connectors to handle this specific value. Maybe we should query this capability and only call this code if it's implemented?

Reply:
So - convert it to an optional operation?

Reply:
I thought we agreed in the previous discussions that time estimates were not part of that PR? I think I'm fine with optional. If not optional, I think that should just be removed from the code at this point of time and a separate PR should be made with the specific changes that add time estimates?